pinot-common/src/test/java/org/apache/pinot/common/metrics/PinotJMXToPromMetricsTest.java

```diff
@@ -0,0 +1,379 @@
+      String.format("%s-%s-%s", TABLE_NAME_WITH_TYPE, KAFKA_TOPIC, PARTITION_GROUP_ID);
+  protected static final String TABLE_STREAM_NAME = String.format("%s_%s", TABLE_NAME_WITH_TYPE, KAFKA_TOPIC);
+  protected static final List<String> EXPORTED_LABELS_FOR_TABLE_NAME_TABLE_TYPE =
+      List.of(LABEL_KEY_TABLE, LABEL_VAL_RAW_TABLENAME, LABEL_KEY_TABLETYPE, LABEL_VAL_TABLETYPE_REALTIME);
+
+  protected static final List<String> EXPORTED_LABELS_FOR_RAW_TABLE_NAME =
+      List.of(LABEL_KEY_TABLE, LABEL_VAL_RAW_TABLENAME);
+
+  protected static final List<String> EXPORTED_LABELS_FOR_CLIENT_ID =
+      List.of(LABEL_KEY_PARTITION, PARTITION_GROUP_ID, LABEL_KEY_TABLE, LABEL_VAL_RAW_TABLENAME, LABEL_KEY_TABLETYPE,
+          TableType.REALTIME.toString(), LABEL_KEY_TOPIC, KAFKA_TOPIC);
+
+  protected static final List<String> EXPORTED_LABELS_FOR_PARTITION_TABLE_NAME_AND_TYPE =
+      List.of(LABEL_KEY_PARTITION, "3", LABEL_KEY_TABLE, LABEL_VAL_RAW_TABLENAME, LABEL_KEY_TABLETYPE,
+          TableType.REALTIME.toString());
+
+  protected static final List<String> EXPORTED_LABELS_FOR_TABLE_TABLETYPE_TASKTYPE =
+      List.of(LABEL_KEY_TABLE, LABEL_VAL_RAW_TABLENAME, LABEL_KEY_TABLETYPE, LABEL_VAL_TABLETYPE_REALTIME,
+          LABEL_KEY_CONTROLLER_TASKTYPE, LABEL_VAL_CONTROLLER_TASKTYPE_CHC);
+
+  protected static final List<String> EXPORTED_LABELS_FOR_TABLENAMEANDTYPE_AND_TASKTYPE =
+      List.of(LABEL_KEY_TABLE, LABEL_VAL_TABLENAME_WITH_TYPE_REALTIME, LABEL_KEY_CONTROLLER_TASKTYPE,
+          LABEL_VAL_CONTROLLER_TASKTYPE_CHC);
+
+  protected static final List<String> EXPORTED_LABELS_FOR_TASK_TYPE_AND_STATUS =
+      List.of(LABEL_KEY_STATUS, LABEL_VAL_STATUS_IN_PROGRESS, LABEL_KEY_CONTROLLER_TASKTYPE,
+          LABEL_VAL_CONTROLLER_TASKTYPE_CHC);
+
+  protected static final List<String> EXPORTED_LABELS_PERIODIC_TASK_TABLE_TABLETYPE =
+      List.of(LABEL_KEY_CONTROLLER_PERIODIC_TASK, LABEL_VAL_CONTROLLER_TASKTYPE_CHC, LABEL_KEY_TABLE,
+          LABEL_VAL_RAW_TABLENAME, LABEL_KEY_TABLETYPE, LABEL_VAL_TABLETYPE_REALTIME);
+
+  /**
+   * For impl, see: https://github.com/prometheus/jmx_exporter/blob/a3b9443564ff5a78c25fd6566396fda2b7cbf216
+   * /jmx_prometheus_javaagent/src/main/java/io/prometheus/jmx/JavaAgent.java#L48
+   * @param pinotComponent the Pinot component to start the server for
+   * @return the corresponding HTTP server on a random unoccupied port
+   */
+  protected HTTPServer startExporter(PinotComponent pinotComponent) {
+    String args = String.format("%s:%s/%s", 0, CONFIG_DIR, PINOT_COMPONENT_CONFIG_FILE_MAP.get(pinotComponent));
+    try {
+      JMXExporterConfig config = parseExporterConfig(args, "0.0.0.0");
+      CollectorRegistry registry = new CollectorRegistry();
+      (new JmxCollector(new File(config._file), JmxCollector.Mode.AGENT)).register(registry);
+      DefaultExports.register(registry);
+      return (new HTTPServerFactory()).createHTTPServer(config._socket, registry, true, new File(config._file));
```

Comment:
nit: parenthesis in `(new HTTPServerFactory())` are not required.

Reply:
BTW: We should keep a reference to the returned server in order to shut them down after tests are executed

Reply:
Ah yes, need to shutdown as part of clean-up, I'll add this.

Reply:
This is done